File: tests/fixtures/language_adapter_symbols/Sample.java
// HAND-DERIVED: standard Java syntax written from language knowledge, not from any adapter regex.
package com.example.inventory;

import java.util.HashMap;
import java.util.Map;

public class Sample {
    private Map<String, Integer> items = new HashMap<>();

    public void addItem(String name, int qty) {
        items.put(name, qty);
    }

    public int totalItems() {
        return items.values().stream().mapToInt(Integer::intValue).sum();
    }
}

interface Describable {
    String describe();
}

File: tests/fixtures/language_adapter_symbols/sample.rs
// HAND-DERIVED: standard Rust syntax written from language knowledge, not from any adapter regex.
use std::collections::HashMap;

pub struct Inventory {
    items: HashMap<String, u32>,
}

impl Inventory {
    pub fn new() -> Self {
        Inventory { items: HashMap::new() }
    }

    pub fn add_item(&mut self, name: &str, qty: u32) {
        self.items.insert(name.to_string(), qty);
    }
}

pub enum Status {
    Active,
    Retired,
}

pub trait Describable {
    fn describe(&self) -> String;
}
